lib/src/types.rs

```diff
@@ -62,20 +62,20 @@ pub enum Stmt {
     TypedStmt(Box<Stmt>, Type),
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub enum StmtExpr {
     Assign(String, Expr),
     New(Type, Vec<Expr>),
     MethodCall(Expr, String, Vec<Expr>),
     TypedStmtExpr(Box<StmtExpr>, Type),
 }
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub enum Expr {
     This,
-    Super,
```

Comment:
Good idea to remove `Super`, I think. Since we have no inheritance or constructor it can't really be used.

Reply:
Yup in CG we'll hardcode the construxtor if needed